src/auth/auth.service.ts

```diff
@@ -1,261 +1,132 @@
+import { SellerEntity } from '@app/saler/entities/saler.entity';
+import { SellerResponseInterface } from '@app/saler/types/sellerResponse.interface';
 import { UserEntity } from '@app/user/entities/user.entity';
-import { UserRole } from '@app/user/enums/userRole.enum';
-import { UserService } from '@app/user/user.service';
-import { EntityCondition } from '@app/utils/types/entityCondition.type';
-import {
-  BadRequestException,
-  Injectable,
-  NotFoundException,
-  UnprocessableEntityException,
-} from '@nestjs/common';
-import { ConfigService } from '@nestjs/config';
+import { Role } from '@app/user/enums/enums';
+import { BadRequestException, HttpException, HttpStatus, Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
 import { compare } from 'bcrypt';
-import * as jwt from 'jsonwebtoken';
+import { sign } from 'jsonwebtoken';
+import { CreateBuyerDto } from 'src/buyer/dto/create-buyer.dto';
+import { Repository } from 'typeorm';
+import { UserLoginDto } from './dto/user-login.dto';
+
+
 
 @Injectable()
 export class AuthService {
   constructor(
-    private readonly mailService: MailService,
-    private readonly configService: ConfigService,
-    private readonly userService: UserService,
-    private readonly buyerService: BuyerService,
-    private readonly sellerService: SellerService,
+    @InjectRepository(BuyerEntity) private readonly buyerRepository: Repository<BuyerEntity>,
+    @InjectRepository(UserEntity) private readonly userRepository: Repository<UserEntity>,
+    @InjectRepository(SellerEntity) private readonly sellerRepository: Repository<SellerEntity>,
   ) {}
-
-  public async registerBuyer(dto: BuyerRegisterDto): Promise<void> {
-    const { email } = dto;
-
-    const user = await this.userService.findOne({ email, emailVerified: true });
-
-    if (user) {
-      throw new EmailAlreadyExistsException(email);
-    }
-
-    await this.buyerService.create(dto);
-
-    await this.sendVerificationMessage({ email, role: UserRole.Buyer });
-  }
-
-  public async registerSellerIndividual(
-    dto: SellerIndividualRegisterDto,
-  ): Promise<void> {
-    const { email } = dto;
-
-    const user = await this.userService.findOne({ email, emailVerified: true });
-
-    if (user) {
-      throw new EmailAlreadyExistsException(email);
-    }
-
-    await this.sellerService.createIndividual(dto);
-
-    await this.sendVerificationMessage({
-      email,
-      role: UserRole.SellerIndividual,
+  async createBuyer(createBuyerDto: CreateBuyerDto): Promise<BuyerEntity> {
+
+    const user = new UserEntity();
+    console.log('user', user)
+    user.email = createBuyerDto.email;
+    user.password = createBuyerDto.password;
+    user.role = Role.Buyer;
+    const buyer = new BuyerEntity();
+    buyer.name = createBuyerDto.name;
+    Object.assign(buyer, createBuyerDto);
+    console.log('newBuyer', buyer)
+            const buyerByEmail = await this.userRepository.findOne({
```

Comment:
looks like the following lines of code have formatting issues

Reply:
✅